students/navdeep/session06/mailroom_part4.py

```diff
@@ -1,75 +1,158 @@
-        showDonorNames(donor_list)
-    elif new_donor.lower() == 'exit':
-        return
-    else:
+    user_donor_option = ""
+    while user_donor_option != 'exit':
+        #user_donor_option = input("1. Type 'new' of a new donor\n"
+         #                         "2. Type 'list' to list the donor names\n"
+          #                        "3. Type 'exit' to return to main menu\n"
+           #                       "Your Selection:  ")
+        #print()
+        user_donor_option = 'NeW'
+        user_donor_option = user_donor_option.strip().lower()
+        if user_donor_option == 'new':
+            return "addNewDonor(donor_dict)"
+        elif user_donor_option == 'list':
+            return "showDonorNames(donor_dict)"
+        elif user_donor_option == 'exit':
+            return
+        else:
+            print("Please select a valid option:")
+
+def addNewDonor(donor_dict):
+    first_name = None
+    last_name = None
+    full_name = None
+    invalid = True
+    while invalid == True:
         try:
-            new_donation = float(input("Enter a new donation amount: "))
+            #first_name = input("Enter first name for donor: ")
+            #last_name = input("Enter last name for donor: ")
+            first_name = "Kelly"
+            last_name = "Pratt"
+            first_name, last_name = first_name.strip(),last_name.strip()
+            if first_name == "" or last_name == "":
+                raise ValueError
         except ValueError as e:
-            print("The amount you entered is an invalid value.")
+            print("The name you entered is invalid. Enter proper first/last name")
+        else:
+            first_name, last_name = first_name.lower(), last_name.lower()
+            first_name,last_name = first_name.title(), last_name.title()
+            invalid = False
+    full_name = first_name + ' ' + last_name
+    #Return statement used for testing purposes
+    return full_name
+    #addNewDonation(donor_dict, full_name)
+
+def addNewDonation(donor_dict, new_donor):
+    """
+    Allows the user to input a new donation for the user. The user must
+    enter a valid amount (integer or float).  The user must
+    also enter a value greater than 0.  If the donor currently exists
+    in the dictionary, the donor's new donation will be appended to the list stored under
+    their name, which is the key in the dictionary. If the donor does
+    not exist in the dictionary, a new key will be created along with a new
+    list for that key.
+    :param donor_list: the dictionary of donors and their donation amounts
+    :param new_donor: the name of the new donor
+    """
+    try:
+        #new_donation = float(input("Enter a new donation amount: "))
+        new_donation = 1000
+        while new_donation < 0:
+            new_donation = float(input("Enter something greater than 0: "))
+    except ValueError as e:
+        print("The amount you entered is an invalid value.")
```

Comment:
You only need to set ValueError to e if you are going to print out e or use that information somehow...